### PR TITLE
Support image path

### LINKED ISSUE #6: Image Support
For Music notifications, for example

### PR DESCRIPTION
**Pull request by danrabbit**
Fixes #6 

Fixes showing album art from Music for example. We badge with the app icon so that we can always tell who is sending notifications. This is important for Mail who sends avatars through

Screenshot from 2019-06-12 20 04 06@2x

Screenshot from 2019-06-12 20 09 04@2x

### CODE REVIEW

**Comment by cassidyjames** [comment]
I'm in favor. I think it's a reasonable feature of the OS to always show what app a notification is coming from, and badging seems like the correct way to do it.

File: src/DBus.vala [hunk 1]
```diff
@@ -91,6 +91,15 @@ public class Notifications.Server : Object {
             app_info = new DesktopAppInfo (desktop_id);
         }
 
+        string? image_path = null;
+        if ((variant = hints.lookup ("image-path")) != null || (variant = hints.lookup ("image_path")) != null) {
+            image_path = variant.get_string ();
+
+            if (!image_path.has_prefix ("/") && !image_path.has_prefix ("file://")) {
```
Comment (danrabbit) [inline src/DBus.vala]: It seems like sometimes apps might try to pass in their icon name here and we don't want that

**Comment by danrabbit** [comment]
@cassidyjames no, Gtk won't clip the image unfortunately. This code was taken from what we're doing in the sound indicator

File: src/Widgets/MaskedImage.vala [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        var image = new Gtk.Image ();
+        image.pixel_size = ICON_SIZE;
+
+        try {
+            var scale = get_style_context ().get_scale ();
+            var pbuf = new Gdk.Pixbuf.from_file_at_size (image_path, ICON_SIZE * scale, ICON_SIZE * scale);
+
+            image.gicon = mask_pixbuf (pbuf, scale);
+        } catch (Error e) {
+            critical ("Unable to mask image: %s",e.message);
```
Comment (donadigo) [inline src/Widgets/MaskedImage.vala]: If we fail to load the image_path for whatever reason (path doesn't exist etc.), perhaps we should fallback to the default icon / app icon?

Comment (danrabbit) [reply src/Widgets/MaskedImage.vala]: Yeah I agree. I'm not certain how to do that though. Should we self destruct this widget and have a null check in Bubble.vala?

**Pull request merged**